app/models/server/models/Subscriptions.js

```diff
@@ -33,6 +33,51 @@ export class Subscriptions extends Base {
 		this.tryEnsureIndex({ prid: 1 });
 	}
 
+	async findByUserIdWithUnreadMessagesCount(userId, options = {}) {
+		return this.model.rawCollection().aggregate([
+			{ $match: { 'u._id': userId } },
+			...(options.sort ? [{ $sort: options.sort }] : []),
+			{
+				$lookup: {
+					from: 'rocketchat_room',
+					localField: 'rid',
+					foreignField: '_id',
+					as: 'room',
+				},
+			},
+			{ $unwind: '$room' },
+			{
+				$lookup: {
+					from: 'rocketchat_message',
+					localField: 'rid',
+					foreignField: 'rid',
+					as: 'unreads',
```

Comment:
Use [pipeline](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/#lookup-join-pipeline) here to filter and get only the unread messages

Reply:
Use pipeline on room lookup as well to get only the `lm` field